packages/opentelemetry-core/test/context/HttpTraceContext.test.ts

```diff
@@ -78,6 +79,23 @@ describe('HttpTraceContext', () => {
       );
       assert.deepStrictEqual(carrier[TRACE_STATE_HEADER], 'foo=bar,baz=qux');
     });
+
+    it('should not set traceparent and tracestate header if instrumentation is suppressed', () => {
+      const spanContext: SpanContext = {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+        traceState: new TraceState('foo=bar,baz=qux'),
+      };
+
+      httpTraceContext.inject(
+        suppressInstrumentation(setSpanContext(ROOT_CONTEXT, spanContext)),
+        carrier,
+        defaultTextMapSetter
+      );
+      assert.deepStrictEqual(carrier[TRACE_PARENT_HEADER], undefined);
```

Comment:
```suggestion
      assert.strictEqual(carrier[TRACE_PARENT_HEADER], undefined);
```
same on next line

Reply:
Curious if there is some advantage to using strict over deep strict?

Reply:
in this usage where one element is `undefined` there is most likely no difference. But in other usecases there is:
```js
assert.deepStrictEqual({}, {})  // throws
assert.strictEqual({}, {}) // doesn't throw
```
At least for me the use of `deepXXX` asserts indicates that deep objects should be compared which is not the case here.

But feel free to ignore my comments regarding this is you feel the opposite.

Reply:
I don't have an opinion at all, was just curious what the reasoning was behind yours.